pyk/src/pyk/cterm/symbolic.py

```diff
@@ -120,9 +120,15 @@ def execute(
 
         state = CTerm.from_kast(self.kore_to_kast(response.state.kore))
         resp_next_states = response.next_states or ()
-        next_states = tuple(CTerm.from_kast(self.kore_to_kast(ns.kore)) for ns in resp_next_states)
+        next_states = tuple(
+            (
+                CTerm.from_kast(self.kore_to_kast(ns.kore)),
+                self.kore_to_kast(ns.rule_predicate) if ns.rule_predicate is not None else None,
+            )
+            for ns in resp_next_states
+        )
 
-        assert all(not cterm.is_bottom for cterm in next_states)
+        assert all(not cterm.is_bottom for cterm, _ in next_states)
```

Comment:
```suggestion
        assert all(not cterm.is_bottom for cterm, _ in next_states)
        assert len(next_states) !=1 or next_states[0].condition is None
```

Reply:
I don't quite understand this one. Line 132 is currently `assert len(next_states) != 1 or response.reason is StopReason.CUT_POINT_RULE` and I think you're suggesting for it to be replaced by `assert len(next_states) !=1 or next_states[0].condition is None`? Are those two equivalent by construction?

Reply:
No, I'm suggesting adding this extra assertion (if it holds).

Reply:
I believe that it is subsumed by the current `response.reason is StopReason.CUT_POINT_RULE`.